tests/test_sag.py

```diff
@@ -0,0 +1,78 @@
+import textwrap
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+import yaml
+import numpy as np
+from astropy.io import fits
+import galsim
+
+import imsim
+
+
+def test_sag():
+    with TemporaryDirectory() as d:
+        config = textwrap.dedent(
+            f"""
+            input:
+                telescope:
+                    file_name: LSST_r.yaml
+                    perturbations:
+                        M2:
+                            shift: [100.e-6, 0.0, 0.0]
+            output:
+                dir: {d}
+                sag:
+                    file_name: sag.fits
+                    nx: 127
+            """
+        )
+
+        config = yaml.safe_load(config)
+        galsim.config.ProcessInput(config)
+        galsim.config.SetupExtraOutput(config)
+        # Need to mock file_num for following method to be happy:
+        config['file_num'] = 0
```

Comment:
This works, but the canonical thing to call to setup file-related things is 
```
galsim.config.SetupConfigFileNum(config, 0, 0, 0)
```
which does a bit that could be necessary in some contexts.